File: code_examples/3-Structured_Problem_Solving/9-Select_Case/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module checks a student's grade
"""


def main() -> None:
    """The main() this function checks a student's grade, returns None."""

    # input
    grade = input("Enter grade mark as a single character(ex: A, B, ...): ")

    # process & output
    # NOTE: This will only work on >= Python 3.10
    match grade.upper():
        case "A":
            print("Excellent!")
        case "B":
            print("Good job!")
        case "C":
            print("Average.")
        case "D":
            print("Poor.")
        case "F":
            print("Fail.")
        case _:
            print("Invalid grade.")

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/9-Select_Case/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program checks your grade

#include <stdio.h>
#include <ctype.h>

int main() {
    // this function checks your grade
    char gradeLevel;  // a single character

    // input
    printf("Enter grade mark as a single character(ex: A, B, ...): ");
    scanf("%c", &gradeLevel);

    // process and output
    // Note you need the break in C or it will move to next
    // line in switch statement if it is true again
    switch (toupper(gradeLevel)) {
        case 'A':
            printf("Excellent!\n");
            break;
        case 'B':
            printf("Good job!\n");
            break;
        case 'C':
            printf("Average.\n");
            break;
        case 'D':
            printf("Poor.\n");
            break;
        case 'F':
            printf("Fail.\n");
            break;
        default:
            printf("Invalid grade.\n");
    }

    printf("\nDone.\n");
    return 0;
}
